### DISCUSSION TITLE
Arbitrary (non-validated) query params in the form of variadic kwargs?

### DISCUSSION THREAD

**Discussion by Datamance** [fastapi/fastapi#8308] [Questions] [answered]
**Description**

I've noticed that I can't define some GET endpoint as such:

```python
@app.get("/items/{item_id}")
def read_item(item_id: int, q: string = None, **kwargs):
    return {"item_id": item_id, "q": q, **kwargs}
```

Meaning, I can't define a route that will take arbitrary query params (i.e., not stipulated by the API schema itself). Am I missing something, or was this a conscious design decision?

**Comment by tiangolo** [comment, marked as answer]
@Datamance you can get the raw `Request` object directly and extract all the query parameters (defined or not): https://fastapi.tiangolo.com/tutorial/using-request-directly/

**Comment by Datamance** [comment]
👏 👏 👏 thank you!

**Comment by tiangolo** [comment]
Great! I'll then close this issue now. But feel free to add me comments or create new issues.

**Comment by inactivist** [comment]
> @Datamance you can get the raw `Request` object directly and extract all the query parameters (defined or not): https://fastapi.tiangolo.com/tutorial/using-request-directly/

For those stumbling on this issue, here's an updated link: [Using the Request Directly](https://fastapi.tiangolo.com/advanced/using-request-directly/)
